freqtrade/configuration.py

```diff
@@ -0,0 +1,199 @@
+"""
+This module contains the configuration class
+"""
+
+import json
+
+from typing import Dict, List, Any
+from jsonschema import Draft4Validator, validate
+from jsonschema.exceptions import ValidationError, best_match
+
+from freqtrade.constants import Constants
+from freqtrade.logger import Logger
+
+
+class Configuration(object):
+    """
+    Class to read and init the bot configuration
+    Reuse this class for the bot, backtesting, hyperopt and every script that required configuration
+    """
+    def __init__(self, args: List[str], do_not_init=False) -> None:
+        self.args = args
+        self.logging = Logger(name=__name__)
+        self.logger = self.logging.get_logger()
+        self.config = None
+
+    def load_config(self) -> Dict[str, Any]:
+        """
+        Extract information for sys.argv and load the bot configuration
+        :return: Configuration dictionary
+        """
+        self.logger.info('Using config: %s ...', self.args.config)
+        config = self._load_config_file(self.args.config)
+
+        # Add the strategy file to use
+        config.update({'strategy': self.args.strategy})
+
+        # Load Common configuration
+        config = self._load_common_config(config)
+
+        # Load Backtesting
+        config = self._load_backtesting_config(config)
+
+        # Load Hyperopt
+        config = self._load_hyperopt_config(config)
+
+        return config
+
+    def _load_config_file(self, path: str) -> Dict[str, Any]:
+        """
+        Loads a config file from the given path
+        :param path: path as str
+        :return: configuration as dictionary
+        """
+        try:
+            with open(path) as file:
+                conf = json.load(file)
+        except FileNotFoundError:
+            self.logger.critical(
+                'Config file "%s" not found. Please create your config file',
+                path
+            )
+            exit(0)
```

Comment:
I don't think `exit(0)` is a good action here, because it's very deep inside the Configuration class and `0` would indicate a sucessful exit. I'm in favor of raising an exception and catching it on a outer level